test/e2e/knative_serving_test.go

```diff
@@ -36,17 +37,35 @@ func TestKnativeServing(t *testing.T) {
 				t.Fatalf("Deployment %s is not ready: %v", deployment, err)
 			}
 		}
-		// Check the desired scale of deployments in the ingress namespace.
-		ingressDeployments := []string{"net-kourier-controller", "3scale-kourier-gateway"}
-		for _, deployment := range ingressDeployments {
-			if err := test.CheckDeploymentScale(caCtx, servingNamespace+"-ingress", deployment, haReplicas); err != nil {
-				t.Fatalf("Failed to verify default HA settings: %v", err)
+
+		// If FULL_MESH is true, net-istio is used instead of net-kourier.
+		mesh := os.Getenv("FULL_MESH")
+		if mesh == "true" {
+			// Check the desired scale of ingress deployments in the knative serving namespace
+			for _, deployment := range []string{"net-istio-controller", "net-istio-webhook"} {
+				if err := test.CheckDeploymentScale(caCtx, servingNamespace, deployment, haReplicas); err != nil {
+					t.Fatalf("Failed to verify default HA settings for %q: %v", deployment, err)
+				}
 			}
-		}
-		// Check the status of deployments in the ingress namespace.
-		for _, deployment := range ingressDeployments {
-			if _, err := test.WithDeploymentReady(caCtx, deployment, servingNamespace+"-ingress"); err != nil {
-				t.Fatalf("Deployment %s is not ready: %v", deployment, err)
+			// Check the status of ingress deployments in the knative serving namespace
+			for _, deployment := range []string{"net-istio-controller", "net-istio-webhook"} {
```

Comment:
would it make sense to combine both loops to a single loop as you always have to iterate over all deployments, so if the number of deployments changes you are not in danger to forgot to update one loop ?

Reply:
or put the deployments in a slice and reuse it for both loops, like it is done below for the kourier case

Reply:
net-kourier deployments are deployed in `knative-serving-ingress` ns while `net-istio` is deployed in `knative-serving` ns.
So we need to different `test.CheckDeploymentScale(caCtx, servingNamespace)` vs `test.CheckDeploymentScale(caCtx, servingNamespace+"-ingress")`, deployment, haReplicas)` and it cannot make the combined loop simple, I think.

Reply:
But give me a sec. I will try...

Reply:
Thank you! Updated.